### PR TITLE
Enable tests and test cmdline -> config file processing

### PR DESCRIPTION
**Pull request by derek-keeler**
To enable testing, let's ensure our process from command-line to runtime-config has some clear tests in place.

_(This will aid in any move from `argparse` to other command-line parsers later)_

Note that in the course of doing this work, a few new issues were raised!
- #34
- #35
- #36
- #37
- #38

### CODE REVIEW

File: .vscode/recommended.settings.json [hunk 1]
```diff
@@ -4,5 +4,8 @@
     "editor.formatOnSave": true,
     "python.testing.pytestEnabled": true,
     "python.linting.flake8Enabled": true,
-    "python.linting.mypyEnabled": true
+    "python.linting.mypyEnabled": true,
+    "python.testing.pytestArgs": [],
+    "python.testing.unittestEnabled": false,
+    "python.testing.nosetestsEnabled": false
```
Comment (eniklas) [inline .vscode/recommended.settings.json:10]: I should just copy these settings into my `settings.json` right?

File: gamatrix-gog.py [hunk 2]
```diff
@@ -265,10 +266,7 @@ def set_multiplayer_status(game_list, cache):
         game_list[k]["max_players"] = max_players
 
 
-if __name__ == "__main__":
-    logging.basicConfig(level=logging.INFO)
-    log = logging.getLogger()
-
+def parse_cmdline(argv: List[str]) -> Any:
```
Comment (eniklas) [inline gamatrix-gog.py:269]: We just starting using `typing` at work last week 🥇

File: gamatrix-gog.py [hunk 3]
```diff
@@ -309,10 +307,24 @@ def set_multiplayer_status(game_list, cache):
...
+    args = parse_cmdline(sys.argv[1:])
+
     if args.debug:
         log.setLevel(logging.DEBUG)
 
+    # in case we want to see our command line shenanigans in a vebose session:
+    log.debug("Command line arguments:")
+    log.debug(sys.argv)
+    log.debug("Arguments after parsing:")
+    log.debug(args)
```
Comment (eniklas) [inline gamatrix-gog.py]: Nit, but mind putting these on 2 lines, e.g.:

`log.debug(f"Command line arguments: {sys.argv}")`

You can put a newline in there if you want to retain that in the output. Not a big deal if you want to keep these as is.

**Review by eniklas — APPROVED** [review]
Tiny bit of pedantry about saving 2 lines, but nothing I'll fight for :-)

This is awesome, our first tests!

File: .vscode/recommended.settings.json (hunk 1 above)
Comment (derek-keeler) [reply .vscode/recommended.settings.json:10]: If you like!

If you do, you will get this to appear in VSCode:

image

File: gamatrix-gog.py (hunk 3 above)
Comment (derek-keeler) [reply gamatrix-gog.py]: Done.

**Pull request merged**